### PR TITLE
Programmatically verify file upload success for daily report

### PR DESCRIPTION
**Pull request by nicholaspai**
[Slack API](https://api.slack.com/methods/files.upload) is expected to respond `{ok:true,...}` if the upload was successful, we can verify that. We'll print out file type and size if successful.

New dockerfile requirement:
- Installs [jq](https://stedolan.github.io/jq/) which is `sed` for JSON data

Other update
- Add report to `.gitignore` in case they are not properly deleted (i.e. if script stops for whatever reason)
- log response.json in a local file we can use to debug

Signed-off-by: Nick Pai <npai.nyc@gmail.com>

### CODE REVIEW

File: reporters/SendSlackReport.sh [hunk 1]
```diff
@@ -25,9 +25,23 @@ echo "${reportOutput}" >$fileName
...
+# Verify that the 'ok' property of the response data is "true".
+# API documentation can be found here: https://api.slack.com/methods/files.upload
+uploadSuccess=$(cat $responseFileName | jq '.ok')
+if [ "$uploadSuccess" = true ] ; then
+    fileType=$(cat $responseFileName | jq '.file.pretty_type')
+    fileSize=$(cat $responseFileName | jq '.file.size')
+    echo "Success! Uploaded a new" $fileType "file ("$fileSize "bytes)"
+else
+    error=$(cat $responseFileName | jq '.error')
+    echo "Failed to upload file! Error description:" $error
```
Comment (chrismaree) [inline reporters/SendSlackReport.sh:43]: Prehaps add a `exit 1` here to indicate exit with error.

Comment (chrismaree) [reply reporters/SendSlackReport.sh:43]: this will help with cloud run catching errors coming from the process execution.

Comment (nicholaspai) [reply reporters/SendSlackReport.sh:43]: smart

[the remark this answers is not in the document]
Comment (mrice32) [reply reporters/SendSlackReport.sh:43]: lol @chrismaree same comment same time #brainlink

[the remark this answers is not in the document]
Comment (nicholaspai) [reply reporters/SendSlackReport.sh:43]: cool. Should i also explicitly exit with code 0 if we get to end of script? or is that default

[the remark this answers is not in the document]
Comment (nicholaspai) [reply reporters/SendSlackReport.sh:43]: updated

Comment (nicholaspai) [reply reporters/SendSlackReport.sh:43]: updated

**Pull request merged**